### DISCUSSION TITLE
How to see debugging info for AR

### DISCUSSION THREAD

**Discussion by timhc22** [pmndrs/xr#198] [General] [answered]
Is it possible to get debugging info displaying in a fixed position in AR View?
I was hoping something like this would work (where debug contains various data):

```
      <Canvas linear>
        <XR>
            ...
        </XR>
        <Html className='debugInfo'>
          {JSON.stringify(debug)}
        </Html>
      </Canvas>
```

If there is another way to see what the output of `hitMatrix` and `hit` are that would be incredible.
Or even if there is some kind of recommended AR simulator for development for the browser..

**Comment by rekliner** [comment]
the <Text> object from @react-three/drei could accomplish what you want but it will probably take some legwork to limit what text is presented and how is handles overflow.  [Take a look at this VR example](https://codesandbox.io/s/three-fiber-xr-headset-camera-bounds-box-tin2ko?file=/components/CameraBoundsBox.js:174-191) that updates camera position to a text element with each frame.

If you are open to a flat screen based solution and using a Quest 2 [you can follow these instructions](https://developer.oculus.com/documentation/web/browser-remote-debugging/) to get a developer console window on your screen that follows the headset ar session.  That way you can use standard console.log statements.  However this is probably not your platform since Quest 2 only enabled in-browser AR last week and it still doesn't support hit testing.

**Reply by timhc22** [reply]
Cool I'll check that thanks. Annoyingly, most of the examples I find (for obvious time based reasons), are pre v5. The main issue I need to fix is figuring out the new syntax for tracking hitTest with the XR component (have made another post about this). Will try your suggestion though thanks

**Comment by CodyJasonBennett** [comment, marked as answer]
Have you tried the WebXR emulator or connecting an Android device via USB debugging? Chrome should have ADB developer tools you can access from `chrome://inspect#devices`.

**Reply by timhc22** [reply]
Ah the emulator, good shout, hadn't seen that one. The AR button doesn't seem to be clickable though, maybe I need to add some extra params.. 🤔
